### PR TITLE
feat(scripts): Auto-set cluster from aws login information

### PR DESCRIPTION
**Pull request by AndesKrrrrrrrrrrr**
## Summary by CodeRabbit

- **New Features**
	- Introduced a dynamic `CLUSTER` variable for improved configuration based on the AWS account context.
	- Added a new `.envrc.kube` file to set up Kubernetes context based on the AWS profile.
	- Implemented a timing function in `.envrc` to track loading times for environment files.
- **Bug Fixes**
	- Enhanced error handling in scripts to improve robustness and prevent unexpected exits.
- **Chores**
	- Modified `.gitignore` to handle private environment configuration files and IDE settings.

### CODE REVIEW

File: .envrc [hunk 1]
```diff
@@ -19,4 +19,6 @@ fi
 
 # Podman compatibility
 # NOTE: Exits direnv if user is using docker.
-source_env_if_exists .env.podman
+source_env_if_exists .envrc.podman
+
+source_env_if_exists .envrc.kube
```
Comment (eirikurn) [inline .envrc:24]: We had an issue once where envs weren't loaded after .envrc.podman. See the comment above.

Probably makes sense to fix the podman script to not exit envrc.

File: scripts/run-proxies.sh [hunk 2]
```diff
@@ -10,6 +10,13 @@ if [[ -n "${DEBUG:-}" || -n "${CI:-}" ]]; then set -x; fi
 : "${RESTART_MAX_RETRIES:=3}"
 : "${LOCAL_PORT:=}"
 : "${DRY:=}"
+: "${CLUSTER:=$(
+  echo "CLUSTER environment variable is not set. Setting using \`kubectl\` and \`aws sts get-caller-identity\`" >&2
+  kubectx "$(kubectx | grep "$(aws sts get-caller-identity | jq -r '.Account')")" >&2
+  kubectl config current-context | grep -oP '(?<=/).*'
+)}"
```
Comment (eirikurn) [inline scripts/run-proxies.sh:17]: Is this adding a new dependency for devs to run proxies? Then we should update the proxy docs and announce it on slack.

Comment (AndesKrrrrrrrrrrr) [reply scripts/run-proxies.sh:17]: Valid point. I'm gonna look into whether I can do this with pure `kubectl`

File: .envrc (hunk 1 above)
Comment (AndesKrrrrrrrrrrr) [reply .envrc:24]: Yup, changed to `return` instead.

Working on generalization [here](https://github.com/island-is/island.is/pull/16458)

**Review by eirikurn — APPROVED** [review]
LGTM 👍

**Comment by robertaandersen** [comment]
```
direnv: loading ~/Source/island.is/.envrc
direnv: loading ~/Source/island.is/.envrc.kube
Setting kube context...
Switched to context "arn:aws:eks:eu-west-1:821090935708:cluster/shared-cluster01".
grep: invalid option -- P
usage: grep [-abcdDEFGHhIiJLlMmnOopqRSsUVvwXxZz] [-A num] [-B num] [-C[num]]
	[-e pattern] [-f file] [--binary-files=value] [--color=when]
	[--context[=num]] [--directories=action] [--label] [--line-buffered]
	[--null] [pattern] [file ...]
environment:400: pop_var_context: head of shell_variables not a function context
environment:17: pop_var_context: head of shell_variables not a function context
direnv: error exit status 2
```

**Review by robertaandersen — APPROVED** [review]
LGTM

**Pull request merged**